src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java

```diff
@@ -0,0 +1,206 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.controllers.CcdCallbackController;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.dm.DocumentManagementUploadService;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.helper.CaseSearcher;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.helper.CcdCaseCreator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.helper.EnvelopeMessager;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.ScannedDocument;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdApi;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.events.CreateExceptionRecord;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.logging.appinsights.SyntheticHeaders;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.http.HttpHeaders.AUTHORIZATION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.helper.CaseDataExtractor.getScannedDocuments;
+
+@SpringBootTest
+@ActiveProfiles("nosb") // no servicebus queue handler registration
+class CreateCaseTest {
+
+    @Value("${test-url}")
+    private String testUrl;
+
+    @Value("${document_management.url}")
+    private String documentManagementUrl;
+
+    @Value("${document_management.context-path}")
+    private String dmContextPath;
+
+    @Autowired
+    private CcdApi ccdApi;
+
+    @Autowired
+    private CcdCaseCreator ccdCaseCreator;
+
+    @Autowired
+    private CaseSearcher caseSearcher;
+
+    @Autowired
+    private EnvelopeMessager envelopeMessager;
+
+    @Autowired
+    private DocumentManagementUploadService dmUploadService;
+
+    @Autowired
+    private CcdAuthenticatorFactory ccdAuthenticatorFactory;
+
+    private String dmUrl;
+
+    @BeforeEach
+    void setup() throws Exception {
+
+        dmUrl = dmUploadService.uploadToDmStore(
+            "Certificate.pdf",
+            "documents/supplementary-evidence.pdf"
+        );
+    }
+
+    @Test
+    public void should_create_case_upon_exception_record() throws Exception {
```

Comment:
may be `should_create_case_from_valid_exception_record` ?

Reply:
Done